src/components/settings/PinInput.tsx

```diff
@@ -0,0 +1,126 @@
+import classNames from 'classnames';
+import { type Dispatch, type SetStateAction, useCallback, useRef } from 'react';
+import { Pin } from '../../types';
+
+export default function PinInput({
+  pin,
+  className,
+  onChange,
+  type = 'password',
+}: {
+  pin: Pin | undefined;
+  onChange: Dispatch<SetStateAction<Pin | undefined>>;
+  className?: string;
+  type?: 'text' | 'password';
+}) {
+  // Create refs for each input
+  const firstInputRef = useRef<HTMLInputElement>(null);
+  const secondInputRef = useRef<HTMLInputElement>(null);
+  const thirdInputRef = useRef<HTMLInputElement>(null);
+  const fourthInputRef = useRef<HTMLInputElement>(null);
+
+  const changeDigit = useCallback(
+    (
+      digit: 'first' | 'second' | 'third' | 'fourth',
+      value: string | undefined,
+    ) => {
+      const parsedValue = value ? Number.parseInt(value) : undefined;
+
+      onChange(
```

Comment:
not sure what's going on but once I enter two digits quite fast, it does not move to another field..

Reply:
it is because there was already a number entered, for example 1, and if you type 1 again in it, it doesn't trigger onChange because nothing was changed, using onInput instead to fix 👍